src/FriendPanel.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useRef } from 'react';
+import { flushSync } from 'react-dom';
+import './FriendPanel.css';
+
+export default function FriendPanel({selectFriend, selectedFriend, message, setMessage, history, setHistory}) {
+    const [friends, setFriends] = useState([]);
+    const [newFriend, setNewFriend] = useState('');
+    const selectedRef = useRef(null);
+  
+    const handleChangeFriend = (event) => {
+      setNewFriend(event.target.value);
+    };
+  
+    function addFriend() {
+      if (friends.includes(newFriend)) {
+        flushSync(() => {selectFriend(newFriend)});
+        selectedRef.current.scrollIntoView();
+        console.log('user already exists');
+        console.log(selectedRef.current);
+      } else if(newFriend === '') {
+        console.log('no user entered');
+      } else {
+        const nextFriends = [...friends, newFriend];
+        flushSync(() => {
+            setFriends(nextFriends);
+            selectFriend(newFriend);
+        });
+        selectedRef.current.scrollIntoView();
+      }
+      selectFriend(newFriend);
+    }
+  
+    const friendList = friends.map((FriendName) => {
+      let selected = false;
+      if (FriendName === selectedFriend) {
+        selected = true;
+      }
+  
+      function OpenChat(friendName) {
+        const lastMessageForFriend = history.filter((messageHistory) => {
+            return messageHistory[0] === friendName
+          }).slice(-1);
+        const friendMessageCount = lastMessageForFriend.length;
+        if (message != ''){
```

Comment:
```suggestion
        if (message !== ''){
```
Related to `===` thing we've talked last week, it's also better to use `!==` instead of `!=` for the same reasons. It's more of an advanced read, but you check this (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness) if you wanna learn more about it.